### PR TITLE
FIX issue #7212: TMathText does not display with high value coordinates

### PR DESCRIPTION
**Pull request by couet**
Fix issue #7212.

The TMathText method GetAlignPoint() computes the offset required to implement the text alignment. It does it using
integers (pixels). The value of kMaxPixel was arbitrary fixed to 32000. 
In case of very high range in user coordinates the TMathText position was wrong. There is no reason to fix the kMaxPixel arbitrary to 32000. We can go to the maximal possible integer. That's what this fix is implementing. Doing that the issue #7212 is fixed.

### CODE REVIEW

File: core/gui/inc/GuiTypes.h [hunk 1]
```diff
@@ -365,7 +365,7 @@ struct Rectangle_t {
 // Atoms used for text cut and paste between windows
 const Atom_t kPrimarySelection = 1;  // magic values, must match the ones
 const Atom_t kCutBuffer        = 9;  // in /usr/include/X11/Xatom.h
-const Int_t  kMaxPixel         = 32000;
+const Int_t  kMaxPixel         = 2147483647; // Max value for an int
```
Comment (bellenot) [inline core/gui/inc/GuiTypes.h:368]:
```suggestion
const Int_t  kMaxPixel         = INT_MAX; // Max value for an int
```

**Review by bellenot — CHANGES_REQUESTED** [review]

**Review by bellenot — APPROVED** [review]

**Comment by couet** [comment]
@phsft-bot build

**Comment by couet** [comment]
@phsft-bot build

**Comment by eguiraud** [comment]
Hi, `INT_MAX` [is standard](https://en.cppreference.com/w/cpp/types/climits), but it's defined in `<climits>` not `<limits>`.

**Comment by linev** [comment]
Or one can use `std::numeric_limits<int>::max()`, provided by `#include <limits>`

**Pull request merged**